src/utils/Firestore.ts

```diff
@@ -40,71 +40,69 @@ export default class Firestore {
+
+    const productInStorage = await Promise.all(products.map(p => this.isProductInStorage(p)));
+    products = products.filter((p, i) => !productInStorage[i]);
 
-    if (await this.isProductInStorage(product)) return;
     const family = await CurrentFamily.instance.getCurrentFamily();
 
-    family.storage.push(product);
+    family.storage.push(...products);
+
     await this.db
       .collection('family')
       .doc(family.id)
       .set(family);
   }
 
-  public async removeFromStorage(product: ProductDTO) {
+  public async removeFromStorage(products: ProductDTO[]) {
     const family = await CurrentFamily.instance.getCurrentFamily();
 
-    family.storage = family.storage.filter(candidate => candidate.name != product.name);
+    family.storage = family.storage.filter(candidate =>
+      products.every(deleted => deleted.name !== candidate.name)
+    );
     await this.db
       .collection('family')
       .doc(family.id)
       .set(family);
   }
 
-  public async moveToWasted(product: ProductDTO) {
-    if (product instanceof Product) {
-      product = product.toDTO();
-    }
-    const seconds = new Date().getTime() / 1000;
-    const documents = await this.db
-      .collection('wasteBuckets')
-      .where('familyId', '==', (await CurrentFamily.instance.getCurrentFamily())!.id)
-      .get();
-    const wastedProduct: WastedProduct = {
+  public async moveToWasted(products: ProductDTO[]) {
+    products = products.map(p => (p instanceof Product ? p.toDTO() : p));
+
+    const wastedProducts: WastedProduct[] = products.map(product => ({
       ...product,
-      dateWasted: new firebase.firestore.Timestamp(seconds, 0)
-    };
-    const bucket = documents.docs[0];
-    const updatedWastedList = [...bucket.data().wasted, wastedProduct];
+      dateWasted: firebase.firestore.Timestamp.now()
+    }));
+    const bucket = await CurrentFamily.instance.getWasteBucket();
+    const updatedWastedList = [...bucket.data().wasted, ...wastedProducts];
 
-    await this.db
-      .collection('wasteBuckets')
-      .doc(bucket.id)
-      .update('wasted', updatedWastedList);
+    await bucket.ref.update('wasted', updatedWastedList);
   }
 
-  public async removeFromShoppingList(product: ProductDTO) {
+  public async removeFromShoppingList(products: ProductDTO[]) {
     const family = await CurrentFamily.instance.getCurrentFamily();
 
-    family.shoppingList = family.shoppingList.filter(candidate => candidate.name != product.name);
+    family.shoppingList = family.shoppingList.filter(candidate =>
+      products.every(deleted => deleted.name !== candidate.name)
+    );
+
     await this.db
       .collection('family')
       .doc(family.id)
       .set(family);
   }
 
-  public async addToShoppingList(product: ProductDTO) {
-    if (product instanceof Product) {
-      product = product.toDTO();
-    }
-    if (await this.isProductInShoppingList(product)) return;
+  public async addToShoppingList(products: ProductDTO[]) {
```

Comment:
This is almost identical to the `addToStorage` function - we should consider combining them or extracting another function in the future.

Reply:
oh, wait for it

Reply:
it's gonna be great